### PR TITLE
Tests for Presenters

### PR DESCRIPTION
**Pull request by ksheremet**
relnote: none

### CODE REVIEW

File: core/src/testLocalFirebase/java/org/dasfoo/delern/presenters/DelernMainActivityPresenterTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+
+public class DelernMainActivityPresenterTest extends FirebaseServerUnitTest {
+
+    private final static int FIVE_SEC = 5000;
```
Comment (dotdoom) [inline core/src/testLocalFirebase/java/org/dasfoo/delern/presenters/DelernMainActivityPresenterTest.java]: `TIMEOUT`. We don't want to rename constant every time we change its value, do we?

File: core/src/testLocalFirebase/java/org/dasfoo/delern/presenters/DelernMainActivityPresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        mUser = signIn();
+    }
+
+    @Test
+    @SuppressWarnings("CheckReturnValue")
+    public void createDeck() throws Exception {
+        mUser.save().doOnComplete(() -> {
+            mPresenter.onCreate(mUser);
+            mPresenter.createNewDeck("test");
+            verify(mDelernMainView, timeout(FIVE_SEC)).addCardsToDeck(any(Deck.class));
```
Comment (dotdoom) [inline core/src/testLocalFirebase/java/org/dasfoo/delern/presenters/DelernMainActivityPresenterTest.java]: This code is executed asynchronously, right? So we won't see the result and the test will be marked green even if it fails.

Comment (ksheremet) [inline core/src/testLocalFirebase/java/org/dasfoo/delern/presenters/DelernMainActivityPresenterTest.java]: It doesn't check whether deck was created or not. It checks that views callbacks called correctly.

File: core/src/testLocalFirebase/java/org/dasfoo/delern/presenters/DelernMainActivityPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        // inject the mocks in the test the initMocks method needs to be called.
+        MockitoAnnotations.initMocks(this);
+
+        // Get a reference to the class under test
+        mPresenter = new DelernMainActivityPresenter(mDelernMainView);
+        mUser = signIn();
+    }
+
+    @Test
+    //@SuppressWarnings("CheckReturnValue")
```
Comment (dotdoom) [inline core/src/testLocalFirebase/java/org/dasfoo/delern/presenters/DelernMainActivityPresenterTest.java]: Remove?

**Pull request merged**